File: rb/features.rb
# KonkanRailwayLivePosition SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module KonkanRailwayLivePositionFeatures
  def self.make_feature(name)
    case name
    when "base"
      KonkanRailwayLivePositionBaseFeature.new
    when "test"
      KonkanRailwayLivePositionTestFeature.new
    else
      KonkanRailwayLivePositionBaseFeature.new
    end
  end
end

File: py/features.py
# KonkanRailwayLivePosition SDK feature factory

from feature.base_feature import KonkanRailwayLivePositionBaseFeature
from feature.test_feature import KonkanRailwayLivePositionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KonkanRailwayLivePositionBaseFeature(),
        "test": lambda: KonkanRailwayLivePositionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
